### ISSUE TITLE
Android: Building AtomSampleViewer project fails

### ISSUE THREAD

**Issue by LB-KacperKapusta** [o3de/o3de#5241]
**Describe the bug**
Running the 'gradlew assembleProfile' for the Android generated AtomSampleViewer project results in the build failure.

FAILURE: Build failed with an exception.

```
* What went wrong:
Execution failed for task ':app:externalNativeBuildProfile'.
> Build command failed.
  Error while executing process ninja.exe with arguments {-C C:\o3de-build\android\app\cmake\profile\arm64-v8a AtomSampleViewer.GameLauncher}
  ninja: Entering directory `C:\o3de-build\android\app\cmake\profile\arm64-v8a'
  [1/1602] Building CXX object o3de-atom-sampleviewer-185dc7e4/Gem/Code/CMakeFiles/AtomSampleViewer.Private.Static.dir/Source/Utils/ImGuiProgressList.cpp.o
  [2/1602] Building CXX object o3de-atom-sampleviewer-185dc7e4/Gem/Code/CMakeFiles/AtomSampleViewer.Private.Static.dir/Source/Platform/Android/Utils_Android.cpp.o
  [3/1602] Building CXX object o3de-atom-sampleviewer-185dc7e4/Gem/Code/CMakeFiles/AtomSampleViewer.Private.Static.dir/Source/Passes/RayTracingAmbientOcclusionPass.cpp.o
  FAILED: o3de-atom-sampleviewer-185dc7e4/Gem/Code/CMakeFiles/AtomSampleViewer.Private.Static.dir/Source/Passes/RayTracingAmbientOcclusionPass.cpp.o

```
Note that attempting to build the source engine with the "-DLY-PROJECTS=" set to the AtomSampleViewer project did not solve the issue. 
Please refer to the attached logs for more information.

**Prerequisites**
- AtomSampleViewer project is built for Windows.
- Assets for the Android platform are enabled and processed.
- All the required environment variables are set as described in the instructions: https://o3de.org/docs/user-guide/platforms/android/generating_android_project_windows/

**Steps to reproduce**
1. Generate KeyStore to allow APK signing:
`keytool -genkey -keystore %O3DE_ANDROID_SIGNCONFIG_FILE% -storepass %O3DE_ANDROID_SIGNCONFIG_STORE_PASSWORD% -alias %O3DE_ANDROID_SIGNCONFIG_KEY_ALIAS% -keypass %O3DE_ANDROID_SIGNCONFIG_KEY_PASSWORD% -keyalg RSA -keysize %O3DE_ANDROID_SIGNCONFIG_KEY_SIZE% -validity %O3DE_ANDROID_SIGNCONFIG_VALIDITY_DAYS% -dname %O3DE_ANDROID_DN%`
2. Use Python to run generate_android_project.py:
`%O3DE_ENGINE_PATH%\python\python.cmd %O3DE_ENGINE_PATH%\cmake\Tools\Platform\Android\generate_android_project.py --engine-root %O3DE_ENGINE_PATH% --project-path %O3DE_PROJECT_PATH% --build-dir %O3DE_BUILD_ROOT%\android --third-party-path %USERPROFILE%\.o3de\3rdParty --android-sdk-path %O3DE_ANDROID_SDK_PATH% --android-ndk-version %O3DE_ANDROID_NDK_VERSION% --android-sdk-platform %O3DE_ANDROID_SDK_API_LEVEL% --include-apk-assets --asset-mode %O3DE_ANDROID_ASSET_MODE% --signconfig-store-file %O3DE_ANDROID_SIGNCONFIG_FILE% --signconfig-store-password %O3DE_ANDROID_SIGNCONFIG_STORE_PASSWORD% --signconfig-key-alias %O3DE_ANDROID_SIGNCONFIG_KEY_ALIAS% --signconfig-key-password %O3DE_ANDROID_SIGNCONFIG_KEY_PASSWORD%`
3. Build the Android Project:
`gradlew assembleProfile`

**Expected behavior**
The building finishes successfully.

**Actual behavior**
The building fails.

**Assets required**
https://github.com/o3de/o3de-atom-sampleviewer

**Found in Branch**
Stabilization/2110 (49da85c)
AtomSampleViewer - Stabilization/2110 (ea8a90c)

**Desktop**
Device: PC
OS: Windows
Version: 10
CPU: Intel i5-9600K
GPU: RTX 2070
Memory: 16GB

**Additional context**
https://o3de.org/docs/user-guide/platforms/android/generating_android_project_windows
Log: [gradlew_assembleProfile_stacktrace.txt](https://github.com/o3de/o3de/files/7468117/gradlew_assembleProfile_stacktrace.txt)

**Comment by LB-KacperKapusta** [comment]
@Adi-Amazon, the described issue does no longer reproduce after pulling the latest ASV Stabilization/2110 (064fbe5) and reverifying the provided steps.

Please note that after the building is completed and with the Launcher being deployed to an Android device, we were not able to launch the ASV GameLauncher, as described in https://github.com/o3de/o3de/issues/5305.

**Comment by moraaar** [comment]
With the latest code from o3de (development) 42d5368618a0872d2dcbebaf104613d004bbe19f and asv (development) 0a97e27b0afec5dc38d02a5de2eee058d46d3a9f android platform builds correctly. Closing this bug.

**Comment by LB-KacperSzczepanski** [comment]
Issue verified as fixed on build Development (e4abc2a)
